.github/workflows/docker-build-push.yml

```
@@ -1,38 +1,80 @@
-name: Build and Push Docker
+name: Build, Render, and Push
 
 # Controls when the action will run. Triggers the workflow on push
-# events only for the master branch, and only when the Dockerfile changes
+# events only for the master branch
 on:
   push:
     branches: [ master ]
-    paths: [ docker/Dockerfile ]
 
 jobs:
-  # This workflow contains a single job called "build-push"
-  build-push:
+  # This workflow contains a single job called "build-all"
+  build-all:
     # The type of runner that the job will run on
     runs-on: ubuntu-latest
     # Steps represent a sequence of tasks that will be executed as part of the job
     steps:
       # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
-      - uses: actions/checkout@v2
-      # Login to dockerhub
+      - name: checkout
+        uses: actions/checkout@v2
+        with:
+          # get the full repo history
+          fetch-depth: 0
+
+      # Test if Dockerfile has changed 
+      # sets steps.check_docker.outputs.changed to 1 if the dockerfile has changed, 0 otherwise
+      - name: Check Dockerfile for changes
+        id: check_docker
+        env:
+          BEFORE: ${{ github.event.before }}
+        run: |
+          git diff-index --name-only $BEFORE > changes.txt
+          if grep "docker/Dockerfile" changes.txt ; then
+            echo "Dockerfile changed"
+            echo "::set-output name=changed::1"
+          else
+            echo "No change to Dockerfile"
+            echo "::set-output name=changed::0"
+          fi
+
+      # Login to Dockerhub
       - name: Login to DockerHub
+        if: steps.check_docker.outputs.changed == 1
         uses: docker/login-action@v1
         with:
           username: ${{ secrets.DOCKER_ID }}
           password: ${{ secrets.DOCKER_PASSWORD }}
       # set up Docker build
       - name: Set up Docker Buildx
+        if: steps.check_docker.outputs.changed == 1
         uses: docker/setup-buildx-action@v1
       # Build docker image (We are not using caching here to force a clean build)
       - name: Build and Push Docker image
+        if: steps.check_docker.outputs.changed == 1
         uses: docker/build-push-action@v2
         with:
           push: true
+          load: true
           context: docker
           file: docker/Dockerfile
           tags: ccdl/refinebio-examples:latest
+      # download data
+      - name: Download data
+        run: bash scripts/download-data.sh
+      # render pages with snakemake in the docker image
+      - name: Render all pages to html
+        run: |
+          docker run \
+          --mount type=bind,target=/home/rstudio,source=$PWD \
+          ccdl/refinebio-examples \
+          snakemake --cores 2 --forceall
```

```suggestion
          snakemake --cores 2 --forceall
```
I think at some point you had advised bumping it up to `4`, but I don't know what GHA's machinery is like so I'm suspecting you are keeping it at `2` for a reason.

---

Yeah, I think we only get one cpu on github actions. 2 threads may speed things up a touch with little risk, so I did that.